nbt.json

```diff
@@ -121,8 +135,25 @@
       },
       {
         "name": "value",
-        "type": ["nbtSwitch",{"type":"type"}]
+        "type": [
+          "nbtSwitch",
+          {
+            "type": "type"
+          }
+        ]
       }
     ]
+  ],
+  "anonOptionalNbt": [
+    "optionalNbtType",
+    {
+      "tagType": "anonymousNbt"
+    }
+  ],
+  "optionalNbt": [
+    "optionalNbtType",
+    {
+      "tagType": "nbt"
+    }
   ]
```

Comment:
These don't have to be here, but can be put into all of the the protocol.json's if needed. `optionalNbtType` is the base type that can take a custom `tagType` to read from the stream. I kept `optionalNbt` naming here to refer to the default `nbt` tag just for compatibility with current protocol.json's

Reply:
I think it makes sense to have them here